perceval/utils/statevector.py

```diff
@@ -239,6 +239,22 @@ def m(self):
     def n_max(self):
         return self._n_max
 
+    @staticmethod
+    def tensor_product(svd1, svd2, prob_threshold: float = 0):
+        """
+        Compute the tensor product of two SVDistribution with an optional probability threshold
+        """
+        if len(svd1) == 0:
+            return svd2
+        new_dist = SVDistribution()
+        for sv1, proba1 in svd1.items():
+            for sv2, proba2 in svd2.items():
+                if proba1 * proba2 < prob_threshold:
+                    continue
+                sv = sv1 * sv2
+                new_dist[sv] += proba1 * proba2
+        return new_dist
```

Comment:
Ne faut-il pas renormaliser la SVD, si on a filtré des SV ?

Reply:
That's right, but I didn't normalize here to avoid unnecessary computations. I've normalized after the source has added all the noise in the distribution.